pull-requests.md

```diff
@@ -3,6 +3,7 @@
 WP-CLI follows a pull request workflow for changes to its code (and documentation). Whether you want to fix a bug or implement a new feature, the process is pretty much the same:
 
 0. [Search existing issues](https://github.com/issues?utf8=%E2%9C%93&q=is%3Aopen+sort%3Aupdated-desc+org%3Awp-cli); if you can't find anything related to what you want to work on, open a new issue in the appropriate repository so that you can get some initial feedback.
+    1. Opening an issue before submitting a pull request helps us provide architectural and implementation guidance before you spend too much time on the code.
 1. Fork the repository you'd like to modify, either the framework or one of the command packages.
```

Comment:
> Numbers and formatting are off.

It was deliberate:

![image](https://user-images.githubusercontent.com/36432/28216675-5a28da58-6867-11e7-9335-88bee9f45533.png)

> Maybe add: "... before submitting a pull request allows us to discuss whether the intended change is a good fit in the first place and helps us provide architectural ..."

I think this is reasonably implied in "get some initial feedback" and "implementation guidance". Some features may be implemented as pull requests against existing projects, and other features may be best implemented as standalone packages.